core/src/main/java/io/substrait/extension/ExtensionCollector.java

```diff
@@ -93,11 +109,14 @@ public void addExtensionsToPlan(Plan.Builder builder) {
               .build();
       extensionList.add(decl);
     }
-
-    builder.addAllExtensionUris(uris.values());
-    builder.addAllExtensions(extensionList);
+    return new SimpleExtensions(uris, extensionList);
   }
 
+  @Desugar
```

Comment:
It might be possible to pass this Jabel validation https://github.com/bsideup/jabel/blob/master/jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/RecordsRetrofittingTaskListener.java#L484